File: Practica 5/Estadisticas/POO.py
import math

class Estadisticas:
    def __init__(self, datos):
        self.datos = datos

    def promedio(self):
        suma = 0
        for num in self.datos:
            suma += num
        return suma / len(self.datos)

    def desviacion(self):
        promedio = self.promedio()
        sumas = 0

        for num in self.datos:
            sumas += math.pow(num - promedio, 2)

        return math.sqrt(sumas / (len(self.datos) - 1))

    def mostrar(self):
        print("El promedio es %.5f" % self.promedio())
        print("La desviacion estandard es %.5f" % self.desviacion())

def main():
    entrada = input()
    numeros = list(map(float, entrada.split()))

    e1 = Estadisticas(numeros)
    e1.mostrar()

if __name__ == "__main__":
    main()

File: Practica 5/Estadisticas/POO.java
package Estadisticas;
import java.util.Scanner;
public class POO {
    private double[] datos;

    public POO(double[] datos) {
        this.datos = datos;
    }
    
   
    
    public double promedio(){
        double suma=0;
        for(double num: datos){
            suma += num;
        }
        return suma/datos.length;
    }
    
    public double desviacion(){
        double promedio =promedio();
        double sumas=0;
        
        for(double num: datos){
            sumas += Math.pow(num-promedio, 2);
            
        }
        return  Math.sqrt(sumas / (datos.length-1));
    }
    
    public void mostrar(){
        System.out.printf("El promedio es %.5f%n",promedio());
        System.out.printf("La desviacion estandard es %.5f%n",desviacion());
    }
    public static void main(String[] args) {
        Scanner scanner =new Scanner(System.in);
        double[] numeros=new double[10];
        
        for(int i=0;i<10;i++){
            numeros[i]=scanner.nextDouble();
        }
        
        POO e1=new POO(numeros);
        
        e1.mostrar();
    }
    
}

